event/event.pb.go

```diff
@@ -91,3 +95,27 @@ var fileDescriptor_2d17a9d3f0ddf27e = []byte{
 	0x6c, 0x2a, 0x1c, 0x6f, 0x5e, 0x4f, 0x5a, 0x77, 0x67, 0xbf, 0x7f, 0xd6, 0xd6, 0xbc, 0xf8, 0x6f,
 	0xc1, 0xd3, 0xf7, 0x00, 0x00, 0x00, 0xff, 0xff, 0x32, 0x9b, 0xad, 0xff, 0xf6, 0x01, 0x00, 0x00,
 }
+
+func (this *Event) DataHash() [crypto_sha256.Size]byte {
+	var buf = new(bytes.Buffer)
+	if this.Hash != nil {
```

Comment:
`Hash` shouldn't be part of the hash calculation.
I just want to say again that I think this generator is overengineering for now and I would highly recommend that we simply manually write these functions where we need them with the right helpers if needed.